packages/job-manager/src/runner/runner.ts

```diff
@@ -0,0 +1,117 @@
+import { JobTaskState } from '../batch/job-task';
+import { PoolMetrics } from '../batch/pool-metrics';
+
+@injectable()
+export class Runner {
+    public runOnce: boolean = false;
+
+    private jobId: string;
+    private jobManagerConfig: JobManagerConfig;
+    private restartAfterTime: Date;
+
+    public constructor(
+        @inject(Batch) private readonly batch: Batch,
+        @inject(Queue) private readonly queue: Queue,
+        @inject(PoolMetrics) private readonly poolMetrics: PoolMetrics,
+        @inject(ServiceConfiguration) private readonly serviceConfig: ServiceConfiguration,
+        @inject(Logger) private readonly logger: Logger,
+    ) {}
+
+    public async run(): Promise<void> {
+        await this.init();
+
+        // tslint:disable-next-line: no-constant-condition
+        while (true) {
+            const poolMetricsInfo = await this.batch.getPoolMetricsInfo();
+            const tasksIncrementCount = this.poolMetrics.getTasksIncrementCount(
+                poolMetricsInfo,
+                this.jobManagerConfig.targetQueuedTasksOverloadRatio,
+            );
+
+            let tasksQueuedCount = 0;
+            if (tasksIncrementCount > 0) {
+                const scanMessages = await this.getMessages(tasksIncrementCount);
+                if (scanMessages.length === 0) {
+                    this.logger.logInfo(`The storage queue '${this.queue.scanQueue}' has no message to process.`);
+
+                    break;
+                }
+
+                tasksQueuedCount = await this.addTasksToJob(scanMessages);
+            }
+
+            this.poolMetrics.poolState.lastTasksIncrementCount = tasksQueuedCount;
+
+            this.logger.logInfo('Pool load statistics', {
+                activeTasks: poolMetricsInfo.load.activeTasks.toString(),
+                runningTasks: poolMetricsInfo.load.runningTasks.toString(),
+                pendingTasks: poolMetricsInfo.load.pendingTasks.toString(),
+                processingSpeedTasksPerMinute: (
+                    (60 / this.jobManagerConfig.tasksIncrementIntervalInSeconds) *
+                    this.poolMetrics.poolState.processingSpeed
+                ).toString(),
+                tasksIncrementCountPerInterval: tasksIncrementCount.toString(),
+                tasksIncrementIntervalInSeconds: this.jobManagerConfig.tasksIncrementIntervalInSeconds.toString(),
+                tasksQueuedCountPerInterval: tasksQueuedCount.toString(),
+            });
+
+            if (this.runOnce) {
+                break;
+            }
+
+            if (moment().toDate() > this.restartAfterTime) {
+                this.logger.logInfo(`Performing scheduled restart after ${this.jobManagerConfig.periodicRestartInHours} hours.`);
+                break;
+            }
+
+            // tslint:disable-next-line: no-string-based-set-timeout
+            await new Promise(r => setTimeout(r, this.jobManagerConfig.tasksIncrementIntervalInSeconds * 1000));
```

Comment:
move this to common package as a function waitForTimeout

Reply:
moved to common